mempool/src/shared_mempool/tasks.rs

```diff
@@ -422,14 +400,20 @@ where
 }
 
 // TODO update counters to ID peers using PeerNetworkId
-fn log_txn_process_results(results: &[SubmissionStatus], sender: Option<PeerId>) {
+fn log_txn_process_results(results: &[SubmissionStatusBundle], sender: Option<PeerNetworkId>) {
     let sender = match sender {
         Some(peer) => peer.to_string(),
         None => "client".to_string(),
     };
-    for (mempool_status, vm_status) in results.iter() {
-        if vm_status.is_some() {
+    for (txn, (mempool_status, maybe_vm_status)) in results.iter() {
+        if let Some(vm_status) = maybe_vm_status {
             // log vm validation failure
+            error!(
+                SecurityEvent::InvalidTransactionMempool,
+                failed_transaction = txn,
+                vm_status = vm_status,
+                sender = sender,
+            );
```

Comment:
why is this an error? it seems to indicate that if Alice sent Bob a blob of vm invalid transactions, Bob would spam his error log? Maybe we should make this a histogram counter with a debug log?